internal/channels/broadcast.go

```diff
@@ -0,0 +1,64 @@
+package channels
+
+import (
+	"context"
+	"sync"
+)
+
+// Broadcaster helps to broadcast messages to multiple subscribed channels.
```

Comment:
can this be replaced with `github.com/alecthomas/types/pubsub`?

Reply:
Unfortunately, I could not get that working without it deadlocking in my use case